### ISSUE TITLE
no dates available

### ISSUE THREAD

**Issue by matheusliraofficial** [jeangnc/us-visa-bot#12]
I believe I'm being banned from making requests to the available dates endpoint, it's always returning "no dates available" in any region, does anyone knows if I need to reduce the requests interval to avoid this kind of issue?

**Comment by mihirdhakan93** [comment]
Same here. I am also being thrown the same issue. No Available Dates. Any help is greatly appreciated.

**Comment by Kindzoku** [comment]
Yeah, it's a ban. Will be unbanned in 3-4 hours. I wrote a Puppeteer solution for same purpose. The safe request frequency is 2 minutes.

**Comment by mosmm1** [comment]
I have waited for 10 hours and nothing change. Anyone had this issue?

**Comment by mosmm1** [comment]
> Yeah, it's a ban. Will be unbanned in 3-4 hours. I wrote a Puppeteer solution for same purpose. The safe request frequency is 2 minutes.

I have been banned, waited for 10 hours with no luck. Any idea how to get unbanned?

**Comment by mosmm1** [comment]
> Yeah, it's a ban. Will be unbanned in 3-4 hours. I wrote a Puppeteer solution for same purpose. The safe request frequency is 2 minutes.

How to add it to the bot?

**Comment by Jeetpal1** [comment]
So just use the following sleep() function instead which calls the function every 2 - 5 minutes randomly:
`// sleeps randomly from 2 to 5 minutes
function sleep() {
  const randomDelay = Math.floor(Math.random() * (300 - 120 + 1) + 120); // Generate random number between 120 and 300 (2 to 5 minutes)
  return new Promise((resolve) => {
    setTimeout(resolve, randomDelay * 1000); // Convert seconds to milliseconds
  });
}`
